### PR TITLE
[Form recognizer] Add support for field value type currency

### PR DESCRIPTION
**Pull request by samvaity**
This PR adds support for  for field value type currency.

### CODE REVIEW

**Review by joshfree — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/DocumentField.java [hunk 1]
```diff
@@ -326,6 +323,25 @@ void setValueMap(Map<String, DocumentField> valueMap) {
...
+    public CurrencyValue getValueCurrency() {
+        return valueCurrency;
+    }
+
+    /**
+     * Set the valueCurrency property: Currency value.
+     *
+     * @param valueCurrency the valueCurrency value to set.
+     */
+    public DocumentField setValueCurrency(CurrencyValue valueCurrency) {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/DocumentField.java:340]: other setters return `void` but this setter returns the object itself and public?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/DocumentFieldType.java [hunk 2]
```diff
@@ -42,6 +42,9 @@
     /** Static value object for DocumentFieldType. */
     public static final DocumentFieldType MAP = fromString("object");
 
+    /** Static value currency for DocumentFieldType. */
+    public static final com.azure.ai.formrecognizer.models.DocumentFieldType CURRENCY = fromString("currency");
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/DocumentFieldType.java:46]: Full address of DocumentFieldType?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/CurrencyValue.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/CurrencyValue.java:1]: autogenerated?

**Comment by samvaity** [comment]
Closed with https://github.com/Azure/azure-sdk-for-java-pr/pull/1361

**Pull request closed without merging**